SQLite/Core/ConnectionPool.swift

```diff
@@ -151,44 +153,41 @@ public final class ConnectionPool {
         
         var borrowed : BorrowedConnection!
         
-        repeat {
+        dispatch_semaphore_wait(connectionSemaphore, DISPATCH_TIME_FOREVER)
```

Comment:
The diff didn't come through very well on the changes below. All I did was remove the `repeat` loop and add this semaphore wait before the `dispatch_sync` call.